classes/Workspace.php

```diff
@@ -55,11 +55,59 @@ public function createFolders(): void
     {
         foreach ($this->paths as $path) {
             if (!file_exists($path) && !@mkdir($path)) {
-                throw new Exception($this->translator->trans('Unable to create directory %s', [$path]));
+                throw new IOException($this->translator->trans('Unable to create directory %s', [$path]));
             }
             if (!is_writable($path)) {
-                throw new Exception($this->translator->trans('Cannot write to the directory. Please ensure you have the necessary write permissions on "%s".', [$path]));
+                throw new IOException($this->translator->trans('Cannot write to the directory. Please ensure you have the necessary write permissions on "%s".', [$path]));
+            }
+
+            $this->createPhpIndex($path);
+        }
+    }
+
+    /**
+     * @throws IOException
+     */
+    public function createPhpIndex(string $directoryPath): void
+    {
+        $filePath = $directoryPath . DIRECTORY_SEPARATOR . 'index.php';
+        if (!file_exists($filePath)) {
+            if (!copy(_PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'index.php', $filePath)) {
+                throw new IOException($this->translator->trans('Unable to create file %s', [$filePath]));
+            }
+        }
+    }
+
+    /**
+     * @throws IOException
+     */
+    public function createHtAccess(string $modulePath): void
+    {
+        $filePath = $modulePath . DIRECTORY_SEPARATOR . '.htaccess';
+        $content = <<<HTACCESS
+<IfModule mod_rewrite.c>
+    RewriteEngine On
+
+    RewriteCond %{REQUEST_URI} ^.*/autoupgrade/logs/ [NC]
+    RewriteCond %{REQUEST_URI} !\.txt$ [NC]
+    RewriteRule ^ - [F]
+
+    RewriteRule ^ - [L]
+</IfModule>
+HTACCESS;
+        if (!file_exists($filePath)) {
+            if (!file_put_contents($filePath, $content)) {
```

Comment:
we use the method `Filesystem::dumpFile` and let it generate the IOException in case of error?